extensions/positron-python/python_files/positron/positron_ipykernel/tests/test_positron_jedilsp.py

```diff
@@ -300,3 +303,26 @@ def test_publish_diagnostics(source: str, messages: List[str]):
     actual_messages = [diagnostic.message for diagnostic in actual_diagnostics]
     assert actual_uri == uri
     assert actual_messages == messages
+
+
+def test_close_notebook_cell_clears_diagnostics():
+    # See: https://github.com/posit-dev/positron/issues/4160
+    code = """\
+---
+echo: false
+---
+"""
+    params = DidCloseTextDocumentParams(
+        TextDocumentIdentifier("vscode-notebook-cell:/foo.ipynb#W0sZmlsZQ%3D%3D")
+    )
+    server = mock_server(params.text_document.uri, code, {})
+
+    positron_did_close_diagnostics(server, params)
+
+    # Wait for the diagnostics to be published
```

Comment:
I was hesitant to add such a slow test. Might be worth making the debounce behavior configurable and setting it to zero during tests.

Reply:
Also worth considering whether we actually want diagnostics to be debounced?

That was inherited from upstream (https://github.com/pappasam/jedi-language-server/issues/241) but I'm not sure which is a better UX.

Reply:
>making the debounce behavior configurable and setting it to zero during tests

Yeah, maybe this is a good idea, or at least something like 100ms so we know there is some debouncing?

>whether we actually want diagnostics to be debounced

I like a bit of debouncing. Maybe 1s is too much though. Probably depends widely on the user but maybe 0.5s is okay?

Reply:
Cool, I've sped up the tests.

0.5 and 1s feel close enough that I think we can leave it as is for now and see if we get any user feedback / strong opinions.